aspnetcore/blazor/fundamentals/signalr.md

```diff
@@ -650,27 +650,43 @@ By default, components are prerendered on the server before the client connectio
 
 Monitor inbound circuit activity using the <xref:Microsoft.AspNetCore.Components.Server.Circuits.CircuitHandler.CreateInboundActivityHandler%2A> method on <xref:Microsoft.AspNetCore.Components.Server.Circuits.CircuitHandler>. Inbound circuit activity is any activity sent from the browser to the server, such as UI events or JavaScript-to-.NET interop calls.
 
-For example, you can use a circuit activity handler to detect if the client is idle:
+For example, you can use a circuit activity handler to detect if the client is idle and log its circuit ID (<xref:Microsoft.AspNetCore.Components.Server.Circuits.Circuit.Id?displayProperty=nameWithType>):
 
 ```csharp
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.Extensions.Options;
+using Timer = System.Timers.Timer;
+
 public sealed class IdleCircuitHandler : CircuitHandler, IDisposable
 {
-    readonly Timer timer;
-    readonly ILogger logger;
+    private Circuit? currentCircuit;
+    private readonly ILogger logger;
+    private readonly Timer timer;
 
-    public IdleCircuitHandler(IOptions<IdleCircuitOptions> options, 
-        ILogger<IdleCircuitHandler> logger)
+    public IdleCircuitHandler(ILogger<IdleCircuitHandler> logger, 
+        IOptions<IdleCircuitOptions> options)
     {
-        timer = new Timer();
-        timer.Interval = options.Value.IdleTimeout.TotalMilliseconds;
-        timer.AutoReset = false;
+        timer = new Timer
+        {
+            Interval = options.Value.IdleTimeout.TotalMilliseconds,
+            AutoReset = false
+        };
+
         timer.Elapsed += CircuitIdle;
         this.logger = logger;
     }
 
     private void CircuitIdle(object? sender, System.Timers.ElapsedEventArgs e)
     {
-        logger.LogInformation("{Circuit} is idle", nameof(CircuitIdle));
+        logger.LogInformation("{CircuitId} is idle", currentCircuit?.Id);
+    }
+
+    public override Task OnCircuitOpenedAsync(Circuit circuit, 
+        CancellationToken cancellationToken)
+    {
+        currentCircuit = circuit;
+
+        return base.OnCircuitOpenedAsync(circuit, cancellationToken);
```

Comment:
It's not necessary to call the base method here - it doesn't have a meaningful implementation and just returns `Task.CompletedTask`.

Reply:
If I don't return, I get an error that the `OnCircuitOpenedAsync` isn't returning a value. How is it set up without returning the base method?

Reply:
Make it just return a completed `Task`?

Reply:
Yep, returning `Task.CompletedTask` is the right move!